api/src/permissions/modules/process-ast/process.ts

```diff
@@ -0,0 +1,45 @@
+import type { PermissionsAction } from '@directus/system-data';
+import type { Accountability } from '@directus/types';
+import type { AST } from '../../../types/ast.js';
+import { fetchPermissions } from '../../lib/fetch-permissions.js';
+import { fetchPolicies } from '../../lib/fetch-policies.js';
+import type { Context } from '../../types.js';
+import { fieldMapFromAst } from './lib/field-map-from-ast.js';
+import { injectCases } from './lib/inject-cases.js';
+import type { FieldMap } from './types.js';
+import { collectionsInFieldMap } from './utils/collections-in-field-map.js';
+import { validatePath } from './utils/validate-path.js';
+
+export interface ProcessAstOptions {
+	ast: AST;
+	action: PermissionsAction;
+	accountability: Accountability | null;
+}
+
+export async function processAst(options: ProcessAstOptions, context: Context) {
+	if (!options.accountability || options.accountability.admin) {
+		// TODO this should still go through validatePath to check for non-existing
+		// collections/fields
+		return options.ast;
+	}
+
+	// FieldMap is a Map of paths in the AST, with each path containing the collection and fields in
+	// that collection that the AST path tries to access
+	const fieldMap: FieldMap = fieldMapFromAst(options.ast, context.schema);
+	const collections = collectionsInFieldMap(fieldMap);
+
+	const policies = await fetchPolicies(options.accountability, context);
+
+	const permissions = await fetchPermissions(
+		{ action: options.action, policies, collections, accountability: options.accountability },
+		context,
+	);
+
+	for (const [path, { collection, fields }] of fieldMap.entries()) {
+		validatePath(path, permissions, collection, fields);
```

Comment:
`validatePath` errors when querying with the default `fields=*` when there is no permission for the related collection in relational fields.

The previous behaviour "ignored" the relational field without permission, by not including it within the AST.

Previous logic that is now in `parse-fields.ts` with the permission check removed: 
https://github.com/directus/directus/blob/9e91405677bb1803c6159605914b2015bd2cdcee/api/src/utils/get-ast-from-query.ts#L246-L249

Reply:
Fixed in https://github.com/directus/directus/pull/22384/commits/3993ea5a06533ae41dfb8712f9adf4672f19c30f